packages/app/src/vis-packs/nexus/utils.ts

```diff
@@ -230,31 +230,31 @@ export function getSilxStyle(
   }
 }
 
-export function getDatasetLabel(
+export function getDatasetInfo(
   dataset: Dataset,
   attrValuesStore: AttrValuesStore
-): string {
-  const longName = attrValuesStore.getSingle(dataset, 'long_name');
-  if (longName && typeof longName === 'string') {
-    return longName;
-  }
-
-  const units = attrValuesStore.getSingle(dataset, 'units');
-  if (units && typeof units === 'string') {
-    return `${dataset.name} (${units})`;
-  }
-
-  return dataset.name;
+): DatasetInfo {
+  const rawLongName = attrValuesStore.getSingle(dataset, 'long_name');
+  const longName =
+    rawLongName && typeof rawLongName === 'string' ? rawLongName : undefined;
+
+  const rawUnits = attrValuesStore.getSingle(dataset, 'units');
+  const units = rawUnits && typeof rawUnits === 'string' ? rawUnits : undefined;
+
+  return {
+    label: longName || (units ? `${dataset.name} (${units})` : dataset.name),
+    unit: units,
+  };
```

Comment:
Seemed silly/convoluted to retrieve the units first and pass them to `getDatasetLabel` or to get the units twice or make a function for that, so I made one function for retrieving both info.